src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react"
+import "./dropdown.scss"
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+
+
+export enum DropdownSelectType {
+    CHECKBOX = 'CHECKBOX',
+    LIST = 'LIST',
+}
+export interface DropdownProps {
+    options: { label: string, value: string }[];
+    defaultText: string;
+    type?: DropdownSelectType;
+    boxStyle?: {};
+    listStyle?: {};
+    disabled?: boolean;
+}
+
+const Dropdown = (props: DropdownProps) => {
+    // Set default to list if no style selected
+    let defaultDropdownType: DropdownSelectType = props.type as DropdownSelectType || DropdownSelectType.LIST;
+
+    const [isOpen, setIsOpen] = useState(true)
+    const [selectedOption, setSelectedOption] = useState(props.defaultText)
+
+    const updateSelection = (e: React.MouseEvent<HTMLLIElement>) => {
+        // console.log(e.target.value)
+        if (e.target instanceof HTMLElement) {
+            const targetVal = e.target.getAttribute('value')
+            console.log(targetVal)
+            setSelectedOption(e.target.textContent || '')
+            setIsOpen(false)
+        }
+    }
+
+    useEffect(() => {
```

Comment:
Why set `isOpen` to true in the state variable declaration and then override that with a mount effect setting it to false?

Reply:
typo :3